Carbon/Functions/Grant-Permission.ps1

```diff
@@ -258,6 +257,44 @@ function Grant-CPermission
                     return
                 }
 
+                if( -not ($certificate.PrivateKey | Get-Member 'CspKeyContainerInfo') )
+                {
+                    $privateKeyFileName = $certificate.PrivateKey.Key.UniqueName
+                    # See https://docs.microsoft.com/en-us/windows/win32/seccng/key-storage-and-retrieval
+                    $keyStoragePaths =         @(
+                        "$($env:AppDATA)\Microsoft\Crypto", 
+                        "$($env:ALLUSERSPROFILE)\Application Data\Microsoft\Crypto\SystemKeys", 
+                        "$($env:WINDIR)\ServiceProfiles\LocalService\AppData\Roaming\Microsoft\Crypto\Keys", 
+                        "$($env:WINDIR)\ServiceProfiles\NetworkService\AppData\Roaming\Microsoft\Crypto\Keys", 
+                        "$($env:ALLUSERSPROFILE)\Application Data\Microsoft\Crypto",
+                        "$($env:ALLUSERSPROFILE)\Microsoft\Crypto"
+                    )
+
+                    $privateKeyFiles = $keyStoragePaths | Get-ChildItem -Recurse -Force -ErrorAction Ignore -Filter $privateKeyFileName
+                    if( -not $privateKeyFiles )
+                    {
+                        $msg = "Failed to find the private key file for certificate ""$($Path)"" (subject: $($certificate.Subject); " +
+                                "thumbprint: $($certificate.Thumbprint); expected file name: $($privateKeyFileName)). This is most " +
+                                "likely because you don't have permission to read private keys, or we''re not looking in the right " +
+                                "places. According to <a href='https://docs.microsoft.com/en-us/windows/win32/seccng/key-storage-and-retrieval' target='_blank' rel='noreferrer'>Microsoft docs</a>, " +

```

Comment:
Use Markdown syntax on the two links in this error message. These error messages won't be shown as HTML:

* [Microsoft docs](url)
* [submit an issue/bug report](url)